File: VB-Examples/01_MIMIE Message/ExtractMessageContents.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Email
Imports System.IO

Namespace ExtractMessageContents
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			Dim mail As MailMessage = MailMessage.Load("..\..\..\..\..\..\Data\Sample.eml")
			Dim sb As New StringBuilder()
			sb.AppendLine("From:")
			sb.AppendLine(mail.From.Address)
			sb.AppendLine("To:")
			For Each toAddress As MailAddress In mail.To
				sb.AppendLine(toAddress.Address)
			Next toAddress
			sb.AppendLine("Subject:")
			sb.AppendLine(mail.Subject)
			If mail.BodyHtml IsNot Nothing Then
				sb.AppendLine(mail.BodyHtml)
			ElseIf mail.BodyText IsNot Nothing Then
				sb.AppendLine(mail.BodyText)
			End If
			File.WriteAllText("ExtractedContents.txt", sb.ToString())
            MessageBox.Show("Completed")
		End Sub
	End Class
End Namespace

File: CS-Examples/01_MIMIE Message/ExtractMessageContents.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Email;
using System.IO;

namespace ExtractMessageContents
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            MailMessage mail = MailMessage.Load(@"..\..\..\..\..\..\Data\Sample.eml");
            StringBuilder sb = new StringBuilder();
            sb.AppendLine("From:");
            sb.AppendLine(mail.From.Address);
            sb.AppendLine("To:");
            foreach (MailAddress toAddress in mail.To)
            {
                sb.AppendLine(toAddress.Address);
            }
            sb.AppendLine("Subject:");
            sb.AppendLine(mail.Subject);
            if (mail.BodyHtml != null)
            {
                sb.AppendLine(mail.BodyHtml);
            }
            else if (mail.BodyText != null)
            {
                sb.AppendLine(mail.BodyText);
            }
            File.WriteAllText("ExtractedContents.txt", sb.ToString());
            MessageBox.Show("Completed");
        }
    }
}